packages/babel-parser/src/parser/statement.js

```diff
@@ -838,14 +847,16 @@ export default class StatementParser extends ExpressionParser {
     directives: ?(N.Directive[]),
     topLevel: boolean,
     end: TokenType,
+    afterBlockParse?: (hasStrictModeDirective: boolean) => void,
   ): void {
+    const octalPositions = [];
     let parsedNonDirective = false;
-    let oldStrict;
-    let octalPosition;
+    let oldStrict = null;
 
     while (!this.eat(end)) {
-      if (!parsedNonDirective && this.state.containsOctal && !octalPosition) {
-        octalPosition = this.state.octalPosition;
+      // Track octal literals that occur before a "use strict" directive.
+      if (!parsedNonDirective && this.state.octalPositions.length) {
+        octalPositions.push(this.state.octalPositions);
```

Comment:
```suggestion
        octalPositions.push.apply(octalPositions, this.state.octalPositions);
```
or
```suggestion
        octalPositions.push(...this.state.octalPositions);
```

And you can avoid using `.flat()`